colearn_grpc/utils.py

```diff
@@ -0,0 +1,80 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+import pickle
+from math import ceil
+from typing import Iterator
+
+from colearn.ml_interface import Weights
+from colearn_grpc.proto.generated.interface_pb2 import WeightsPart
+
+WEIGHTS_PART_SIZE_BYTES = 4 * 10 ** 6
+
+
+def encode_weights(w: Weights) -> bytes:
+    return pickle.dumps(w)
+
+
+def decode_weights(w: bytes) -> Weights:
+    return pickle.loads(w)
+
+
+def iterator_to_weights(request_iterator: Iterator[WeightsPart], decode=True) -> Weights:
```

Comment:
can we add prometheus timers to this function? Just have to create timer1=prometheus_client.Summary object, and decorate this function with  @timer1.time()